tests/cpp/test_alias.cpp

```diff
@@ -1395,4 +1395,36 @@ TEST_F(AliasTest, SegmentMetaOps) {
   }
 }
 
+TEST_F(AliasTest, StackedAliases) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeContigConcreteTensor({2, 3});
```

Comment:
Note, this code change is actually only beneficial when we have dynamic shape, otherwise, the slice in the example isn't creating segmentation as expected.